unstructured_ingest/v2/processes/connectors/notion/connector.py

```diff
@@ -0,0 +1,342 @@
+import typing as t
+from dataclasses import dataclass
+from time import time
+from typing import Any, Generator, List, Optional, Set, Tuple
+from uuid import UUID
+
+from pydantic import Field, Secret
+
+from unstructured_ingest.error import SourceConnectionError
+from unstructured_ingest.utils.dep_check import requires_dependencies
+from unstructured_ingest.v2.interfaces import (
+    AccessConfig,
+    ConnectionConfig,
+    Downloader,
+    DownloaderConfig,
+    DownloadResponse,
+    FileData,
+    FileDataSourceMetadata,
+    Indexer,
+    IndexerConfig,
+    SourceIdentifiers,
+)
+from unstructured_ingest.v2.logger import logger
+from unstructured_ingest.v2.processes.connector_registry import SourceRegistryEntry
+
+if t.TYPE_CHECKING:
+    from unstructured_ingest.v2.processes.connectors.notion.client import Client as NotionClient
+
+
+NOTION_API_VERSION = "2022-06-28"
+CONNECTOR_TYPE = "notion"
+
+
+class NotionAccessConfig(AccessConfig):
+    notion_api_key: str = Field(description="Notion API key")
+
+
+class NotionConnectionConfig(ConnectionConfig):
+    access_config: Secret[NotionAccessConfig]
+
+
+class NotionIndexerConfig(IndexerConfig):
+    page_ids: Optional[List[str]] = Field(
+        default=None, description="List of Notion page IDs to process"
+    )
+
+    database_ids: Optional[List[str]] = Field(
+        default=None, description="List of Notion database IDs to process"
+    )
+    recursive: bool = Field(
+        default=False, description="Recursively process child pages and databases"
+    )
+
+    def __post_init__(self):
+        if self.page_ids:
+            self.page_ids = [str(UUID(p.strip())) for p in self.page_ids]
```

Comment:
We should be able to use pydantics support for UUIDs here now that the configs are base models rather than dataclasses: [usage/types/uuids/](https://docs.pydantic.dev/2.0/usage/types/uuids/)